library(Biostrings)
library(msa)
library(seqinr)
library(ape)

asio <- readDNAStringSet("C:/Users/emmag/OneDrive/School Work/2024-2025/SPRING 2024/ST GR Bioinformatics - BSC 6932/Bioinformatics_Spring2024/Megascops asio cytb MSA/m_asio_asio_cytb.fasta")
hasbroucki <- readDNAStringSet("C:/Users/emmag/OneDrive/School Work/2024-2025/SPRING 2024/ST GR Bioinformatics - BSC 6932/Bioinformatics_Spring2024/Megascops asio cytb MSA/m_asio_hasbroucki_cytb.fasta")
mccallii <- readDNAStringSet("C:/Users/emmag/OneDrive/School Work/2024-2025/SPRING 2024/ST GR Bioinformatics - BSC 6932/Bioinformatics_Spring2024/Megascops asio cytb MSA/m_asio_mccallii_cytb.fasta")
mccallii_isolate <- readDNAStringSet("C:/Users/emmag/OneDrive/School Work/2024-2025/SPRING 2024/ST GR Bioinformatics - BSC 6932/Bioinformatics_Spring2024/Megascops asio cytb MSA/m_asio_mccallii_isolate_cytb.fasta")
naevius <- readDNAStringSet("C:/Users/emmag/OneDrive/School Work/2024-2025/SPRING 2024/ST GR Bioinformatics - BSC 6932/Bioinformatics_Spring2024/Megascops asio cytb MSA/m_asio_naevius_cytb.fasta")
m_asio_seqs <- c(asio, hasbroucki, mccallii, mccallii_isolate, naevius)
msa(m_asio_seqs, method=c("ClustalW"))
feb_1st_msa <- msa(m_asio_seqs, method=c("ClustalW"))
print(feb_1st_msa, show="complete")

alphabetFrequency(feb_1st_msa)
m_asio_bases <- alphabetFrequency(feb_1st_msa)
class(m_asio_bases)
sum(m_asio_bases[,2])
sum(m_asio_bases[,3])
sum(m_asio_bases[,1])
sum(m_asio_bases[,4])
(sum(m_asio_bases[,1]))+(sum(m_asio_bases[,4]))+(sum(m_asio_bases[,3]))+(sum(m_asio_bases[,2]))
bases_sum <- (sum(m_asio_bases[,1]))+(sum(m_asio_bases[,4]))+(sum(m_asio_bases[,3]))+(sum(m_asio_bases[,2]))
c_bp <- sum(m_asio_bases[,2])
g_bp <- sum(m_asio_bases[,3])
a_bp <- sum(m_asio_bases[,1])
t_bp <- sum(m_asio_bases[,4])
(c_bp + g_bp)/(bases_sum)
(c_bp + g_bp)/(bases_sum)*100
feb_1st_msa2 <- msaConvert(feb_1st_msa, type="seqinr::alignment")
d <- dist.alignment(feb_1st_msa2, "identity")
as.matrix(d)
m_asio_tree <- nj(d)
plot(m_asio_tree, main="Phylogenetic Tree of M. asio Cytb Mitochondral Sequences")
Biostrings::translate(hasbroucki)
m_asio_phyDat <- msaConvert(feb_1st_msa, type="phangorn::phyDat")
write.phyDat(m_asio_phyDat, "m_asio_alignment.fasta", format="fasta")
